File: sdks/dotnet/src/Models/Branch.cs
using System;
using Newtonsoft.Json;

namespace Puxbay.SDK.Models
{
    public class Branch
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("unique_id")]
        public string UniqueId { get; set; }

        [JsonProperty("address")]
        public string Address { get; set; }

        [JsonProperty("phone")]
        public string Phone { get; set; }

        [JsonProperty("branch_type")]
        public string BranchType { get; set; }

        [JsonProperty("currency_code")]
        public string CurrencyCode { get; set; }

        [JsonProperty("currency_symbol")]
        public string CurrencySymbol { get; set; }

        [JsonProperty("low_stock_threshold")]
        public int LowStockThreshold { get; set; }

        [JsonProperty("created_at")]
        public DateTime? CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public DateTime? UpdatedAt { get; set; }
    }
}

File: sdks/php/src/Models/Branch.php
<?php

namespace Puxbay\Models;

class Branch
{
    public function __construct(
        public readonly string $id,
        public readonly string $name,
        public readonly string $uniqueId,
        public readonly ?string $address = null,
        public readonly ?string $phone = null,
        public readonly string $branchType,
        public readonly string $currencyCode,
        public readonly string $currencySymbol,
        public readonly int $lowStockThreshold,
        public readonly \DateTimeImmutable $createdAt,
        public readonly \DateTimeImmutable $updatedAt,
    ) {
    }

    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'],
            name: $data['name'],
            uniqueId: $data['unique_id'],
            address: $data['address'] ?? null,
            phone: $data['phone'] ?? null,
            branchType: $data['branch_type'],
            currencyCode: $data['currency_code'],
            currencySymbol: $data['currency_symbol'],
            lowStockThreshold: (int) $data['low_stock_threshold'],
            createdAt: new \DateTimeImmutable($data['created_at']),
            updatedAt: new \DateTimeImmutable($data['updated_at']),
        );
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'name' => $this->name,
            'unique_id' => $this->uniqueId,
            'address' => $this->address,
            'branch_type' => $this->branchType,
            'currency_code' => $this->currencyCode,
            'low_stock_threshold' => $this->lowStockThreshold,
        ];
    }
}
